include/xtensor/xbuilder.hpp

```diff
@@ -135,7 +135,7 @@ namespace xt
             }
         };
 
-        template <class T, template<class, class> class F>
+        template <class T, template <class> class K, class F = K<T>>
```

Comment:
The best solution would be to deduce the value type from any callable type (including lambda), but that requires more work. This can be done after this feature has been integrated, though. Let me know which option you prefer.